### PR TITLE
Implemented

### PR DESCRIPTION
**Pull request by De1eF**

### CODE REVIEW

**Review by NadiaHuryk — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -45,6 +48,11 @@
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.springfox</groupId>
+            <artifactId>springfox-boot-starter</artifactId>
+            <version>3.0.0</version>
```
Comment (nacenik) [inline pom.xml:54]: do you really need a version here?

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                .map(productMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping
+    @ApiOperation(value = "Get all products with price between two parameters")
+    public List<ProductResponseDto>
+            getAllByPriceBetween(@RequestParam BigDecimal from,
+                                 @RequestParam BigDecimal to,
+                                 @RequestParam(defaultValue = "0") Integer page,
```
Comment (nacenik) [inline src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]: why don't you add `ApiParam` with descriptions for these parameters?

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Get product by id")
+    public ProductResponseDto get(@PathVariable Long id) {
+        return productMapper.toDto(productService.get(id));
+    }
+
+    @GetMapping("/all")
+    @ApiOperation(value = "Get list of all products")
+    public List<ProductResponseDto>
+            getAll(@RequestParam(defaultValue = "0") Integer page,
```
Comment (nacenik) [inline src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]: why don't you add `ApiParam` with a description for this parameter?

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+@AllArgsConstructor
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+
+    @PostMapping
+    @ApiOperation(value = "Save product to database")
+    public ProductResponseDto save(@RequestBody @Valid ProductRequestDto productRequestDto) {
```
Comment (nacenik) [inline src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]: Add description for request body https://stackoverflow.com/questions/64645528/java-spring-boot-openapi-3-how-to-add-description-for-requestbody

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    @DeleteMapping("/{id}")
+    @ApiOperation(value = "Completely delete a product")
+    public void delete(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @PutMapping("/{id}")
+    @ApiOperation(value = "Update product in the database")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody @Valid ProductRequestDto productRequestDto) {
```
Comment (nacenik) [inline src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]: The same about request body

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                        productMapper.toModel(productRequestDto)));
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Get product by id")
+    public ProductResponseDto get(@PathVariable Long id) {
+        return productMapper.toDto(productService.get(id));
+    }
+
+    @GetMapping("/all")
```
Comment (nacenik) [inline src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]: According to the RESTful rules, an endpoint to get all resources should look like `/products`, not `/products/all`

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution, let's improve it)

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java (hunk 3 above)
Comment (Andriy-dz) [reply src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]: not fix

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java (hunk 2 above)
Comment (Andriy-dz) [reply src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]: not fix

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java (hunk 5 above)
Comment (Andriy-dz) [reply src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]: not fix

**Review by Andriy-dz — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Get product by id")
+    public ProductResponseDto get(@PathVariable Long id) {
+        return productMapper.toDto(productService.get(id));
+    }
+
+    @GetMapping
+    @ApiOperation(value = "Get list of all products")
+    public List<ProductResponseDto>
+            getAll(@ApiParam(value = "current page")
```
Comment (Andriy-dz) [inline src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]:
```suggestion
            getAll(@ApiParam(value = "current page, , default value 0")
```

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                .stream()
+                .map(productMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("price-between")
+    @ApiOperation(value = "Get all products with price between two parameters")
+    public List<ProductResponseDto>
+            getAllByPriceBetween(
+                                 @ApiParam(value = "from price")
```
Comment (Andriy-dz) [inline src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]:
```suggestion
                                 @ApiParam(value = "from price", required = true)
```

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+                                 @RequestParam BigDecimal from,
+                                 @ApiParam(value = "to price")
```
Comment (Andriy-dz) [inline src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]:
```suggestion
                                 @ApiParam(value = "to price", required = true)
```

File: src/main/java/mate/academy/springboot/swagger/controller/ProductController.java (hunk 5 above)
Comment (Andriy-dz) [reply src/main/java/mate/academy/springboot/swagger/controller/ProductController.java]: not fix

**Review by Andriy-dz — CHANGES_REQUESTED** [review]

**Review by Andriy-dz — APPROVED** [review]

**Pull request closed without merging**